### PR TITLE
Removes Istio bootstrap

### PR DESCRIPTION
**Pull request by codefromthecrypt**
The maintenance and side-effects of Istio bootstrap integration have
proven to be the bulk of effort in maintaining this project, and the
bulk of its size. This removes the feature to free us to focus on Envoy,
which better aligns with the primary charter of the project.

Before:
```bash
$ du -k build/bin/*/amd64/getenvoy
77924	build/bin/darwin/amd64/getenvoy
66692	build/bin/linux/amd64/getenvoy
```

Now:
```bash
$ du -k build/bin/*/amd64/getenvoy
20768   build/bin/darwin/amd64/getenvoy
18428   build/bin/linux/amd64/getenvoy
```

### CODE REVIEW

File: pkg/cmd/extension/push/cmd.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/docker/distribution/reference"
+	"github.com/containerd/containerd/reference/docker"
```
Comment (codefromthecrypt) [inline pkg/cmd/extension/push/cmd.go:21]: we don't need to import the docker distribution to verify a reference. NOTE: we still import it indirectly due to oras, but I can manage that in a follow-up PR

File: pkg/extension/workspace/example/envoy/template/defaults.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ var (
 					SocketAddress: &envoycore.SocketAddress{
 						Address: "127.0.0.1",
 						PortSpecifier: &envoycore.SocketAddress_PortValue{
-							PortValue: 9901,
+							PortValue: 9901, // TODO: why default admin port to 9901?
```
Comment (codefromthecrypt) [inline pkg/extension/workspace/example/envoy/template/defaults.go:44]: no idea on this one..

File: pkg/pii/pii.go [hunk 3]
```diff
@@ -121,12 +120,6 @@ func (f Filter) Process(logs []string) []string {
 	return out
 }
 
-// Process process logs with the default filter
-// an empty array of logs and an error instance is returned in the event of an error
-func Process(logs []string) ([]string, error) {
```
Comment (codefromthecrypt) [inline pkg/pii/pii.go:126]: dead code

**Review by codefromthecrypt — COMMENTED** [review]
notes

**Comment by mathetake** [comment]
Others: the rational is not only the maintenance burden, but also the fact that we have only supported 1.2 (https://github.com/tetratelabs/getenvoy/pull/168) in the released getenvoy version.. which itself I think proves that this feature is no longer in use by the end user.

**Comment by codefromthecrypt** [comment]
unit test went from 10m to 3m, but I think most of this is that the linters consume less resources due to the simpler dependency tree.

**Comment by codefromthecrypt** [comment]
yeah it is mostly the check phase and likely less bulk to download which is why PRs can check so much faster now.

Screen Shot 2021-04-15 at 11 20 17 AM
Screen Shot 2021-04-15 at 11 20 49 AM

File: pkg/extension/workspace/example/envoy/template/defaults.go (hunk 2 above)
Comment (mathetake) [reply pkg/extension/workspace/example/envoy/template/defaults.go:44]: Envoy folks usually uses `9901` for admin, but it's just a tradition I think.

**Review by mathetake — APPROVED** [review]
LGTM, but I want @lizan 's view before merging 🙂

**Comment by codefromthecrypt** [comment]
rebased over #179 to bring down another 2MiB

```bash
du -k build/bin/*/amd64/getenvoy
20760	build/bin/darwin/amd64/getenvoy
18420	build/bin/linux/amd64/getenvoy
```

**Comment by codefromthecrypt** [comment]
updated the description with the size difference based on this vs latest master (after rebasing)

**Comment by codefromthecrypt** [comment]
@liamawhite I don't think you've been pinged directly on this, but I think you have been the only user of this feature. I searched github google etc and haven't seen any reference to this not made by an employee. We also know that the last release of getenvoy doesn't support any recent release of istio and no-one complained. Finally, it was marked experimental from the beginning. I hope you agree that for the health of the project it is time to delete this. If not please comment why and help pitch in on the maintenance impact which I could enumerate further on, but choosing to save keystrokes.

**Comment by codefromthecrypt** [comment]
rebased for the 4th time. hopefully we can get an answer soon

**Comment by codefromthecrypt** [comment]
Going to merge this so that we can conserve time. If there are any post-merge discussions about it, we can have them any time between now and our next release, which I don't know the schedule of.

**Pull request merged**